### ISSUE TITLE
compiler/nimeval can't be used twice: fails 2nd time with: `Error: internal error: n is not nil`

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#9180]
/cc @Araq 
env var nimc_D points to compile git repo

```nim
import compiler / [ast, vmdef, vm, nimeval, llstream]
from os import getEnv
proc evalString(code: string, moduleName = "script.nim"):auto=
  let stream = llStreamOpen(code)
  defer: llStreamClose(stream)
  let path = getEnv("nimc_D") & "/lib/"
  var intr = createInterpreter(moduleName, [path])
  defer: destroyInterpreter(intr)
  intr.evalScript(stream)

evalString("echo 1")
evalString("echo 2")
```

```
rnim bugs/nimeval/t03_n_is_not_nil.nim
nim c --nimcache:/tmp/nim//nimcache/ -o:/tmp/nim//app -r bugs/nimeval/t03_n_is_not_nil.nim
1
???(0, 0) Error: internal error: n is not nil
Traceback (most recent call last)
/Users/timothee/git_clone/nim/timn/bugs/nimeval/t03_n_is_not_nil.nim(13) t03_n_is_not_nil
/Users/timothee/git_clone/nim/timn/bugs/nimeval/t03_n_is_not_nil.nim(8) evalString
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/nimeval.nim(115) createInterpreter
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/modules.nim(107) compileSystemModule
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/modules.nim(71) compileModule
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/passes.nim(220) processModule
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/passes.nim(95) closePasses
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/sem.nim(626) myClose
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/msgs.nim(524) internalError
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/msgs.nim(495) liMessage
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/msgs.nim(324) handleError
/Users/timothee/.nimble/pkgs/compiler-0.19.0/compiler/msgs.nim(314) quit
Error: execution of an external program failed: '/tmp/nim//app '

```

**EDIT**
/cc @alaviss 
looks suspiciously similar to [1] which was fixed by @alaviss 

[1] running runNimScript on 2 subsequent nims files causes "Error: internal error: n is not nil" #8235

**Comment by alaviss** [comment]
See #9068

**Comment by timotheecour** [comment]
yes; I saw the thread, a better fix may indeed be to address #9068; but I'll let someone more familiar with that code do the proper fix; until then #9181 fixes this issue so I'd like it to be merged

**Comment by timotheecour** [comment]
I've just renamed https://github.com/nim-lang/Nim/pull/9181 to that effect; I think it's better to close once the test case is merged.
